game/world/managers/objects/units/player/TalentManager.py

```diff
@@ -49,18 +49,37 @@ def send_talent_list(self):
                 if not self.player_mgr.skill_manager.can_ever_use_equipment(spell_item_class, spell_item_subclass_mask):
                     continue
 
+            # Checking magic talents to make sure class can use them.
+            if skill_line_ability.SkillLine == TalentSkillLines.MAGIC_TALENTS:
+                if not TrainerUtils.player_can_learn_magic_talent(training_spell, spell, self.player_mgr):
+                    continue
+
```

Comment:
This could be merged into one method that checks for `skill_line_ability` existence, ItemClass requirements and skill requirements since all of these indicate the talent never being available (`TrainerUtils.player_can_ever_learn_talent`?)